File: Algorithms & Data Structures/Searching Algorithms Implementation JS/BinarySearch (Iterative).js
function binarySearch(array, item) {
    let left = 0,
        right = array.length - 1,
        found = false;
    while (left <= right && !found) {
        let mid = Math.floor((right + left) / 2);
        console.log(mid);
        if (item > array[mid]) {
            left = mid + 1;
        }
        else if (item == array[mid]) {
            found = true;
            return "Found " + item + " at position " + (mid + 1);
        }
        else {
            right = mid - 1;
        }
    }
    return item + " not found!";
}

File: Algorithms & Data Structures/Searching Algorithms Implementation C++/BinarySearch (Iterative).cpp
#include <iostream>
using namespace std;

bool binaryI(int arr[], int size, int item) {
    int left = 0;
    int right = size - 1;
    bool found = 0;
    while(left <= right && !found) {
        int mid = (left + right) / 2;
        if(item > arr[mid]) {
            left = mid + 1;
        }
        else if(item == arr[mid]) {
            found = true;
        }
        else {
            right = mid - 1;
        }
    }
    return found;
}
  
int main() {
  int arr[8] = {5, 1, 8, 2, 7, 4, 3, 6};
}